### DISCUSSION TITLE
Control <table> hide columns - ResizeObserver loop completed with undelivered notifications

### DISCUSSION THREAD

**Discussion by Paolo-lab** [iTwin/iTwinUI#1695] [Q&A]
Hi everyone,
after updating "@itwin/itwinui-react" from version "^1.48.2" to version "^2.12.16", I had to update it because I didn't have "notificationMarker" in my previous version, I'm having problems when in a I control <Table> to hide or display columns,
through "ActionColumn({columnManager: true})" the error it generates is
"ResizeObserver loop completed with undelivered notifications"

what I can say is that the elements in the table are around 200, therefore a fairly small number
unfortunately it gives me references to bundle.js, so it's difficult for me to understand where the problem arises
 
I had the same problem, fortunately solved on a modal

I hope you can give me some help
thanks Paul

**Comment by mayank99** [comment]
Could you provide a minimal reproduction by forking this [sandbox](https://codesandbox.io/s/github/iTwin/iTwinUI/tree/legacy/v2/cra-sandbox?file=/src/App.tsx)? It's hard to understand/debug the problem without the full context.

**Comment by Paolo-lab** [comment]
hello everyone good morning.
thanks for the reply @mayank99, unfortunately it remains complex for me to port the code to sandBox, but I don't know if it's possible, but it seems to me that the problem is with the headers, in version 1.48.2 I had decidedly long headers and the string it was going to hide under the other columns in this version 2.12.16, this doesn't happen, and it creates very tall headers for me

if I go to modify the headers, that is, I make sure that they enter the column, the problem disappears.

I attach two images
1) the first without data


<img width="212" alt="shm 1" src="https://github.com/iTwin/iTwinUI/assets/57446652/f5e8a4a6-1618-4939-acdc-490f38a78761">


2) the second with the data, as you can see the headers become very tall.

<img width="205" alt="shm 2" src="https://github.com/iTwin/iTwinUI/assets/57446652/95e87cc5-ea7f-44b2-a1b6-dadc92da31b9">

a second problem I've encountered is the checkboxes that put a box around the checkbox

thanks for the help Paul

**Reply by mayank99** [reply]
For the header cells becoming very tall, the general advice is to set a `minWidth` in your column object and let it [scroll horizontally](https://itwin.github.io/iTwinUI/legacy/v2/react/?path=/story/core-table--horizontal-scroll). For non-header data cells, there are some other suggestions [here](https://github.com/iTwin/iTwinUI/issues/1227#issuecomment-1520428260).

As for the second problem regarding checkboxes, I believe that is because you may have some stray v1 CSS import on a page that doesn't use v1 components. See [similar comment](https://github.com/iTwin/iTwinUI/pull/1257#issuecomment-1582570990) that might be helpful.
